lib/common/httpclient.c

```diff
@@ -123,9 +123,13 @@ static void on_pool_connect(h2o_socket_t *sock, const char *errstr, void *data,
 
     h2o_iovec_t alpn_proto;
     if (sock->ssl == NULL || (alpn_proto = h2o_socket_ssl_get_selected_protocol(sock)).len == 0) {
+        /* 100% means prior knowledge connect, force h2 */
+        if (client->ctx->protocol_selector.ratio.http2 == 100)
```

Comment:
* Can we add a dedicated boolean flag to `h2o_httpclient_protocol_ratio_t` for this behavior (maybe calling it "force_cleartext_http2")? The safe way of using H2 on cleartext TCP is upgrade, but we are skipping that.
* I think we should exercise this option only when `sock->ssl == NULL`. When TLS is used, we are required to (and can) rely on ALPN negatiation.
* Turning this on only when `ratio.http2 == 100` looks good, we have to do that otherwise `on_pool_connect` might be called given a TCP connection that has already been used for exchanging HTTP/1.1 traffic. Maybe we should add a comment regarding why though.

Reply:
That makes sense! Thank you, what do you think about 0a6b685 ?